extensions/amp-auto-ads/0.1/placement.js

```diff
@@ -196,6 +212,45 @@ export class Placement {
     });
   }
 
+  /**
+   * Gets instructions for the placement in terms of height, width and margins.
+   * If responsive is on, ad should be placed at full viewport width and a
+   * proportionate height, and the margins are adjusted so that the ad edges
+   * stick to both ends of the viewport.
+   * @param {!./ad-network-config.SizeInfoDef} sizing
+   * @return {PlacementDef}
+   * @private
+   */
+  getPlacement_(sizing) {
+    if (isExperimentOn(this.ampdoc.win, 'amp-auto-ads-responsive')) {
```

Comment:
We might need to condition this on type='adsense', or else this would affect anyone using amp auto ads with doubleclick (r.e. https://github.com/ampproject/amphtml/pull/16412). Otherwise please check with @emarchiori whether it's appropriate for responsive sizing to run on doubleclick.

If we do need to limit this to just AdSense, then might be worth adding a member function to AdNetworkConfigDef called isResponsiveSizingEnabled() and then doing the isExperimentOn check in there. That way different networks can experiment with this feature independently as they see fit. (https://github.com/ampproject/amphtml/blob/master/extensions/amp-auto-ads/0.1/ad-network-config.js#L36)